src/js/utils/dom.js

```diff
@@ -279,3 +279,10 @@ export function deviceIsLandscape() {
 
     return isLandscape || (window.orientation === 90 || window.orientation === -90);
 }
+
+// Removes all unallowed HTML tags/attributes from a template string. Useful for user created HTML.
+export function HTMLSafeString(str) {
+    return str.replaceAll(/&|<|>|"|''/gm, function(i) {
+        return '&#'+i.charCodeAt(0)+';';
+     }).replaceAll(/&#60;(\/?)(b|strong|i|em|p|br|ul|ol|li|h.)&#62;/gmi, '<$1$2>');
```

Comment:
Curious where this regex came from and this overall func? Home-grown or perhaps pulled from SO or elsewhere?

Reply:
homegrown. Can break it down if you want